File: developerkhm/AndroidStudy/app/src/main/java/com/example/androidstudy/ui/book/BookFragment.java
package com.example.androidstudy.ui.book;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.example.androidstudy.R;

public class BookFragment extends Fragment {


    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        View root = inflater.inflate(R.layout.fragment_news, container, false);
        final TextView textView = root.findViewById(R.id.text_dashboard);
        textView.setText("book");

        return root;
    }
}

File: developerkhm/AndroidStudy/app/src/main/java/com/example/androidstudy/ui/book/BookFragment.kt
package com.example.androidstudy.ui.book

import android.os.Bundle
import android.view.View
import android.view.inputmethod.EditorInfo
import com.example.androidstudy.R
import com.example.androidstudy.databinding.FragmentBookBinding
import com.example.androidstudy.ui.base.BaseFragment
import com.ironelder.androidarchitecture.view.AdapterSearch

class BookFragment : BaseFragment<FragmentBookBinding>(R.layout.fragment_book) {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    private fun init() {
        setLayout()
    }

    private fun setLayout() {

        binding.searchLayout.searchEditText.onEditorAction(EditorInfo.IME_ACTION_SEARCH)

        binding.searchLayout.searchEditText.setOnEditorActionListener { searchEditText, actionId, event ->
            when (actionId) {
                EditorInfo.IME_ACTION_SEARCH -> {
                    basePresenter.search(searchEditText.text.toString(), typeArray[3])
                }
                else -> false
            }
            true
        }

        binding.searchLayout.resultRecyclerView.adapter =
            AdapterSearch(context, arrayListOf(), "book")
    }
}